File: backend/app/schemas/diagnostic.py
from typing import List

from pydantic import BaseModel, Field


class DiagnosticRequest(BaseModel):
    symptoms: List[str] = Field(..., min_items=1, example=["não liga", "faz barulho"])


class DiagnosticIssue(BaseModel):
    id: int
    title: str
    description: str
    symptoms: List[str]
    solution: str
    follow_up_questions: List[str] = Field(default_factory=list)
    confidence: float = Field(default=0.0, ge=0.0, le=1.0)


class DiagnosticResult(BaseModel):
    query: List[str]
    recommendations: List[DiagnosticIssue]

File: frontend/services/diagnostic.ts
export type DiagnosticIssue = {
  id: number;
  title: string;
  description: string;
  symptoms: string[];
  solution: string;
  follow_up_questions: string[];
  confidence: number;
};

export type DiagnosticResponse = {
  query: string[];
  recommendations: DiagnosticIssue[];
};

export async function diagnoseIssue(symptoms: string[]): Promise<DiagnosticResponse> {
  const response = await fetch("http://127.0.0.1:8000/api/diagnostic/analyze", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ symptoms }),
    cache: "no-store",
  });

  if (!response.ok) {
    const errorBody = await response.text();
    throw new Error(errorBody || "Não foi possível realizar o diagnóstico.");
  }

  return response.json();
}
